### PR TITLE
Added try-except-raise message example

### PR DESCRIPTION
**Pull request by matusvalo**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|     | :bug: Bug fix          |
|     | :sparkles: New feature |
|     | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Related #5953

### CODE REVIEW

File: doc/data/messages/t/try-except-raise/bad.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+try:
+    1 / 0
+except:  # [try-except-raise]
+    raise
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/try-except-raise/bad.py:4]:
```suggestion
try:
    1 / 0
except ZeroDivisionError as e:  # [try-except-raise]
    raise
```

File: doc/data/messages/t/try-except-raise/good.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+try:
+    1 / 0
+except ZeroDivisionError as e:
+    raise ValueError from e
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/t/try-except-raise/good.py:4]:
```suggestion
# The try except might be remove entirely:
1 / 0

# Or another more detailed exception can be raised:
try:
    1 / 0
except ZeroDivisionError as e:
    raise ValueError("The area of the rectangle cannot be zero") from e
```

**Comment by jacobtylerwalls** [comment]
Thanks. Is `details.rst` a good place to summarize the discussion from https://github.com/PyCQA/pylint/issues/5884#issuecomment-1064310365 involving filtering exceptions?

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I agree that we should add a detail.rst, to be clear that there's a legitimate use case for re raising immediately but that this check is smart enough to not have a false positive for it. 

I.e. with the following inheritance tree:
```
      +-- ArithmeticError
      |    +-- FloatingPointError
      |    +-- OverflowError
      |    +-- ZeroDivisionError
```

```python      
try:
    some_calculation(...)
except ZeroDivisionError as e:
    raise
except ArithmeticError:
   pass
```   
will not raise because ZeroDivisionError  inherit from ArithmeticError.

File: doc/data/messages/t/try-except-raise/good.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# The try except might be remove entirely:
```
Comment (jacobtylerwalls) [inline doc/data/messages/t/try-except-raise/good.py:1]:
```suggestion
# The try except might be removed entirely:
```

**Comment by matusvalo** [comment]
Added details.rst. Please review if now it is OK.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**